lib/fn/index.js

```diff
@@ -44,27 +41,19 @@ export function normalizeDevice(options={}) {
  */
 export function normalizeMulti(options={}) {
   if (!Array.isArray(options)) {
+    let {io, pins, ...rest} = options;
     // Loop through the pins property array and make sure each is an object
-    let pins = options.pins.map(pin => {
-      return normalizeIO(pin);
-    });
-    
+    pins = options.pins.map(pin => normalizeIO(pin));
     // If IO is defined on options use it as default
-    if (options.io) {
-      pins.forEach(pin => pin.io = pin.io || options.io);
-      delete options.io;
+    if (io) {
+      pins.forEach(pin => pin.io = pin.io || io);
     }
-    
+
     // Copy each property that is not ```pins``` onto each member of the pins array
-    delete options.pins;
-    pins.map(pin => { return Object.assign(pin, options) });
-    return pins;
+    return pins.map(pin => Object.assign(pin, rest));
   } else {
-    return options.map(pin => {
-      return normalizeIO(pin)
-    });
+    return options.map(pin => normalizeIO(pin));
```

Comment:
The application of rest params awesome. It's cool to see features I haven't really picked up used like this. I'm learning a lot.

Reply:
That's one that I've used quite a bit in React code, especially when forwarding props: 

```js
function f(props) {
  const {a, b, forwardTheseProps} = props;
  return <Xyz { ...forwardTheseProps }>👍 </Xyz>;
}
```